portals/publisher/src/main/webapp/source/src/app/components/GlobalPolicies/Listing/Listing.tsx

```diff
@@ -0,0 +1,1115 @@
+                }
+                else {
+                    APIMAlert.error(intl.formatMessage({
+                        id: 'Error.Deploy.Policy',
+                        defaultMessage: 'Error occurred while deploying the policy',
+                    }));
+                }                
+            })
+            .catch((/* error */) => {
+                // console.error(error);
+                APIMAlert.error(intl.formatMessage({
+                    id: 'Error.Undeploy.Policy',
+                    defaultMessage: 'Error occurred while undeploying the policy',
+                }));
+            })         
+        setLoading(false);
+    }
+
+    /**
+     * Function to undeploy a policy mapping to another enviroment/gateway.
+     * @param {string} gatewayPolicyMappingId - Policy Identifier.
+     * @param {string[]} deployingGateways - Deploying enviroment/gateway.
+     * @param {boolean} deploying - Deploying or undeploying.
+     */
+    const deploy = (gatewayPolicyMappingId: string, deployingGateways: string[]) => {
+        setIsDeployDialogOpen(false);
+        setLoading(true);
+        const deploymentArray = getDeploymentArray(gatewayPolicyMappingId);
+
+        /**
+         * Iterate through the selected gateways and update the deployment array.
+         */
+        const updatedDeploymentArray = deploymentArray.map(({ gatewayLabel, gatewayDeployment }) => {
+            if (deployingGateways.includes(gatewayLabel) && gatewayDeployment === false) {
+                return { gatewayLabel, gatewayDeployment: true };
+            }
+            return { gatewayLabel, gatewayDeployment };
+        });
+        
+        /**
+         * call the backend API and handle the response
+         */
+        const promise = API.engageGlobalPolicy(gatewayPolicyMappingId, updatedDeploymentArray);
+        promise
+            .then((response) => {
+                if (response.status === 200 || response.status === 201) {
+                    setLoading(false);              
+                    APIMAlert.success(intl.formatMessage({
+                        id: 'Policy.Deploy.Successful',
+                        defaultMessage: 'Policy deployed successfully',
+                    }));    
+                    /**
+                     * If successful, add to the state rather than getting from backend.
+                     */
+                    addLabelsToPolicy(
+                        gatewayPolicyMappingId, 
+                        deployingGateways
+                    );                                         
+                }
+                else {
+                    APIMAlert.error(intl.formatMessage({
+                        id: 'Error.Deploy.Policy',
+                        defaultMessage: 'Error occurred while deploying the policy',
+                    }));
+                }                
+            })
+            .catch((/* error */) => {
+                // console.error(error);
```

Comment:
```suggestion
           
```

Reply:
Addressed the required changes by [fddf57f](https://github.com/wso2/apim-apps/pull/515/commits/fddf57fa7308a53094c808e9bf2d2b40ed71b644) and [0ad490c](https://github.com/wso2/apim-apps/pull/515/commits/0ad490c497b08f999babbe521feaf83d029a6698)